### PR TITLE
applies several small optimizations

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- applies several small optimizations

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

**Review by aledeg — APPROVED** [review]

File: app/Utils/feverUtil.php [hunk 1]
```diff
@@ -9,8 +9,11 @@ class FreshRSS_fever_Util {
 	 * @return bool true if the path is writable, false otherwise.
 	 */
 	public static function checkFeverPath(): bool {
-		if (!file_exists(self::FEVER_PATH)) {
-			@mkdir(self::FEVER_PATH, 0770, true);
+		if (!file_exists(self::FEVER_PATH) &&
+			!mkdir($concurrentDirectory = self::FEVER_PATH, 0770, true) &&
+			!is_dir($concurrentDirectory)
+		) {
```
Comment (Alkarex) [inline app/Utils/feverUtil.php:15]: The code before was better: the new one is not safe for concurrency, and also might generate warnings

Comment (Alkarex) [reply app/Utils/feverUtil.php:15]: Ah, sorry, I read a bit too fast. The previous one was not so good either :-)

File: app/install.php [hunk 2]
```diff
@@ -290,15 +290,15 @@ function freshrss_already_installed(): bool {
...
+	} catch (Minz_ConfigurationNamespaceException $e) {
 		return false;
 	}
 
 	// ok, the global conf exists… but what about default user conf?
 	$current_user = $system_conf->default_user;
 	try {
 		FreshRSS_UserConfiguration::init(USERS_PATH . '/' . $current_user . '/config.php');
-	} catch (Minz_FileNotExistException $e) {
+	} catch (Minz_ConfigurationNamespaceException $e) {
```
Comment (Alkarex) [inline app/install.php:301]: This introduced a regression https://github.com/FreshRSS/FreshRSS/issues/6204

File: app/Models/UserConfiguration.php [hunk 3]
```diff
@@ -71,6 +73,7 @@
  */
 final class FreshRSS_UserConfiguration extends Minz_Configuration {
 
+	/** @throws Minz_ConfigurationNamespaceException */
```
Comment (Alkarex) [inline app/Models/UserConfiguration.php:76]: Wrong exception. This one is not possible, and the relevant one was overlooked.
Fixed in https://github.com/FreshRSS/FreshRSS/pull/6205

File: app/Models/SystemConfiguration.php [hunk 4]
```diff
@@ -27,6 +29,7 @@
  */
 final class FreshRSS_SystemConfiguration extends Minz_Configuration {
 
+	/** @throws Minz_ConfigurationNamespaceException */
```
Comment (Alkarex) [inline app/Models/SystemConfiguration.php:32]: Wrong exception. Fixed in https://github.com/FreshRSS/FreshRSS/pull/6205

**Pull request merged**